single-node-k8s.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+set -x
+set -e -o pipefail
+
+ARGUMENT_LIST=(
+    "k8s_version"
+    "ref"
+    "test_type"
+)
+
+opts=$(getopt \
+    --longoptions "$(printf "%s:," "${ARGUMENT_LIST[@]}")help" \
+    --name "$(basename "${0}")" \
+    --options "" \
+    -- "$@"
+)
+ret=$?
+
+if [ ${ret} -ne 0 ]
+then
+    echo "Try '--help' for more information."
+    exit 1
+fi
+
+eval set -- "${opts}"
+
+while true; do
+    case "${1}" in
+    --help)
+        shift
+        echo "Options:"
+        echo "--help|-h                 specify the flags"
+        echo "--ref                     specify the reference of pr"
+        echo "--k8s_version             specify the kubernetes version"
+        echo "--test_type               specify the test type functest|helmtest"
+
+        echo " "
+        echo "Sample Usage:"
+        echo "./single-node-k8s.sh --ref=pull/123/head"
+        exit 0
+        ;;
+    --k8s_version)
+        shift
+        k8s_version=${1}
+        echo "${k8s_version}"
+        ;;
+    --test_type)
+        shift
+        test_type=${1}
+        echo "${test_type}"
+        ;;
+    --ref)
+        shift
+        ref=${1}
+        echo "${ref}"
+        ;;
+    --)
+        shift
+        break
+        ;;
+    esac
+    shift
+done
+
+function set_env() {
+    export GOPATH="/opt/build/go"
+    export GOLANGCI_VERSION="v1.21.0"
+    export GO111MODULE="on"
+    export GOSEC_VERSION="2.0.0"
+    export TEST_COVERAGE="stdout"
+    export GO_METALINTER_THREADS=1
+    export HELM_VERSION="v3.1.2"
+    export VM_DRIVER="none"
+    export MINIKUBE_VERSION="v1.6.0"
+    export CHANGE_MINIKUBE_NONE_USER=true
+    export KUBECONFIG="$HOME/.kube/config"
+    export CEPH_CSI_RUN_ALL_TESTS=true
+}
+
+check_errs()
+{
+  if [ "${1}" -ne "0" ]; then
+    echo "ERROR # ${1} : ${2}"
+    exit ${1}
+  fi
+}
+
+# Set environment variables
+set_env
+
+mkdir -p "$GOPATH"/bin
+# sudo rpm --import 'https://download.ceph.com/keys/release.asc'
+curl https://download.ceph.com/keys/release.asc | sudo apt-key add -
+sudo apt-add-repository "deb https://download.ceph.com/debian-nautilus $(lsb_release -sc) main"
```

Comment:
make sure to install packages with yum/dnf, apt commands are for Debian based distributions